talend-component-maven-plugin/src/main/java/org/talend/sdk/component/maven/BuildComponentM2RepositoryMojoV2.java

```diff
@@ -0,0 +1,291 @@
+import java.io.FileInputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.Writer;
+import java.nio.file.FileVisitResult;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.nio.file.SimpleFileVisitor;
+import java.nio.file.attribute.BasicFileAttributes;
+import java.security.DigestOutputStream;
+import java.security.MessageDigest;
+import java.security.NoSuchAlgorithmException;
+import java.util.Properties;
+import java.util.Set;
+import java.util.zip.ZipEntry;
+import java.util.zip.ZipInputStream;
+
+import org.apache.maven.plugin.MojoExecutionException;
+import org.apache.maven.plugins.annotations.Mojo;
+import org.apache.maven.plugins.annotations.Parameter;
+import org.eclipse.aether.artifact.Artifact;
+import org.eclipse.aether.artifact.DefaultArtifact;
+import org.talend.sdk.component.maven.api.Audience;
+
+@Audience(TALEND_INTERNAL)
+@Mojo(name = "build-repository", defaultPhase = PACKAGE, threadSafe = true, requiresDependencyResolution = TEST)
+public class BuildComponentM2RepositoryMojoV2 extends CarConsumer {
+
+    private static final char[] HEX_CHARS = "0123456789abcdef".toCharArray();
+
+    @Parameter(defaultValue = "true", property = "talend.connectors.write")
+    private Boolean writeVersion;
+
+    @Parameter(defaultValue = "${project.version}", property = "talend.connectors.version")
+    private String version;
+
+    @Parameter(defaultValue = "CONNECTORS_VERSION", property = "talend.connectors.file")
+    private String connectorsVersionFile;
+
+    @Parameter(property = "talend-m2.registryBase")
+    private File componentRegistryBase;
+
+    private Path componentRegistryBasePath; // Can't get a Path directly from a @Parameter...
+
+    @Parameter(property = "talend-m2.root",
+            defaultValue = "${maven.multiModuleProjectDirectory}/target/talend-component-kit/maven")
+    protected File m2Root;
+
+    protected Path m2RootPath; // Can't get a Path directly from a @Parameter...
+
+    @Parameter(property = "talend-m2.clean", defaultValue = "true")
+    private boolean cleanBeforeGeneration;
+
+    @Parameter(defaultValue = "true", property = "talend.repository.createDigestRegistry")
+    private boolean createDigestRegistry;
+
+    @Parameter(defaultValue = "SHA-512", property = "talend.repository.digestAlgorithm")
+    private String digestAlgorithm;
+
+    @Override
+    public void doExecute() throws MojoExecutionException {
+        final Set<Artifact> componentCars = project.getDependencies()
+                .stream()
+                .filter(dependency -> "car".equals(dependency.getType())) // FIXME: car should be extracted
+                .filter(dependency -> "compile".equals(dependency.getScope()))
+                .map(dependency -> new DefaultArtifact(
+                        dependency.getGroupId(),
+                        dependency.getArtifactId(),
+                        dependency.getClassifier(),
+                        "car",
```

Comment:
could use `dependency.getScope()` as it's filtered above instead of "car". Or follow your FIXME ("car" used 3 times).

Reply:
Indeed. 
I'm very surprised I couldn't find any constant in the repository for this as it is an extension created by TCK :thinking: